### PR TITLE
File tesbed (IRS +  EKSS part)

### PR DESCRIPTION
**Pull request by mephenor**
Adjusted the test action a bit, so this runs inside the devcontainer spinnig up all containers in the docker-compose.
Two issues exist currently, which we should fix when extending this to other services:

 - IRS container currently always restarts to deal with Kafka startup. This might mask other issues. This should be done by a healthcheck instead, though I could not yet figure out how to make it work. A variant of the `wait_for_it` script might also be a valid solution.
 
 - Dummy secrets are currently hardcoded. Might be better to inject them from somewhere else.
 
 ```
 Created docker-compose for local file testbed, covering IRS and EKSS for now (GDEV-1420)
 Added simple testcase to check processing of events through the stack (GDEV-1421)
 
 Co-authored-by: Moritz Hahn <Moritz.Hahn@uni-tuebingen.de>
 Co-authored-by: Kersten Breuer <kersten-breuer@outlook.com>
 ```

### CODE REVIEW

File: tests/fixtures/local_fixtures.py [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+from hexkit.providers.s3 import S3Config, S3ObjectStorage  # type: ignore
+
+from tests.fixtures.config import CONFIG
+
+PART_SIZE = 8 * 1024**2
+
+
+@dataclass
+class CheckablePayloadFields:
+    """Fields from recorded event that hold values that can an should be checked"""
```
Comment (KerstenBreuer) [inline tests/fixtures/local_fixtures.py]:
```suggestion
    """Fields from recorded event that hold values that can and should be checked"""
```

File: tests/fixtures/config.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""App config for tests and interaction with the Interrogation Room Service"""
+
+from hexkit.config import config_from_yaml  # type: ignore
+from pydantic import BaseSettings, SecretStr  # type: ignore
+
+
+@config_from_yaml(prefix="tb")
+class Config(BaseSettings):
```
Comment (KerstenBreuer) [inline tests/fixtures/config.py]: We can also inherit from hexkits S3Config and KafkaConfig

File: tests/fixtures/local_fixtures.py [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+    ):
+        await storage.delete_object(
+            bucket_id=CONFIG.inbox_bucket, object_id=CONFIG.object_id
+        )
+    yield storage
+
+
+@pytest_asyncio.fixture
+async def populated_bucket_fixture(
```
Comment (KerstenBreuer) [inline tests/fixtures/local_fixtures.py]: Some of the test logic is here in the fixtures, however, the fixtures should only setup and teardown state required for the test to run.

File: tests/fixtures/local_fixtures.py [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        async with KafkaEventPublisher.construct(config=CONFIG) as publisher:
+            type_ = "file_upload_received"
+            key = CONFIG.object_id
+            topic = "file_uploads"
+            await publisher.publish(
+                payload=populated_bucket_fixture.dict(),
+                type_=type_,
+                key=key,
+                topic=topic,
+            )
```
Comment (KerstenBreuer) [inline tests/fixtures/local_fixtures.py]: You cannot guarantee interrogation results will already be available when the event recorder stops.

**Review by KerstenBreuer — CHANGES_REQUESTED** [review]
Looks good, maybe we can discuss the tests later.

File: tests/fixtures/local_fixtures.py (hunk 1 above)
Comment (MoritzHahn1337) [inline tests/fixtures/local_fixtures.py]:
```suggestion
    """Fields from recorded event that hold values that can and should be checked"""
```

**Review by MoritzHahn1337 — APPROVED** [review]

**Review by MoritzHahn1337 — APPROVED** [review]

**Review by MoritzHahn1337 — APPROVED** [review]

File: .devcontainer/docker-compose.yml [hunk 5]
```diff
@@ -63,10 +66,16 @@ services:
       IRS_EKS_URL: http://ekss/secrets
 
   kafka:
-    image: confluentinc/cp-kafka:latest
+    image: confluentinc/cp-kafka:7.3.0
     restart: always
     depends_on:
       - zookeeper
+    healthcheck:
```
Comment (KerstenBreuer) [inline .devcontainer/docker-compose.yml:73]: Nice

**Review by KerstenBreuer — APPROVED** [review]

**Pull request merged**